### DISCUSSION TITLE
Token being logged by agent container?

### DISCUSSION THREAD

**Discussion by lonix1** [woodpecker-ci/woodpecker#2369] [Ask a Question]
When looking through an agent container's logs, I found this:
```
$ docker logs woodpecker-agent-1
2023/09/04 11:20:39 Token refreshed: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
2023/09/04 11:50:39 Token refreshed: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
2023/09/04 12:20:39 Token refreshed: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```

Logs are not secure; for example they might be redirected to a log aggregation tool. So the token would become compromised.

However that token "xxxx" is not the same as the "personal access token". What is it? Is it maybe a one-time token of some kind (which would be "safe" to log in this case)?

**Comment by anbraten** [comment]
It seems to be coming from the cron scheduler: https://github.com/woodpecker-ci/woodpecker/blob/503252df38154a1ec78dd1371f415018fbec6116/server/cron/cron.go#L129

The token is the oauth token used to authenticate as your user against the forge.

**Reply by lonix1** [reply]
Thanks. The thing is, I wrote "xxxx" above, but actually it is different each time.

So I thought maybe it's a one-time token, i.e. can only be used once?

If so, then it's not ideal to be logged, but nonetheless safe. Otherwise it needs some safety consideration.

**Reply by anbraten** [reply]
IMO we could just remove the actual token from the log line. Would be happy to accept a PR 😉
